### ISSUE TITLE
XATTRs: View backfill for deletes failing with latest build

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#2739]
### Sync Gateway version
1.5.0-487

### Couchbase Server version
5.0.0-3217

### Operating system
CentOS 7

### Config file
```
{
        "adminInterface": "0.0.0.0:4985",
        "compressResponses": false,
        "databases": {
            "db": {
                "bucket": "data-bucket",
                "import_docs": "continuous",
                "password": "password",
                "server": "http://cbs-1.st:8091",
                "unsupported": {
                    "enable_extended_attributes": true
                },
                "username": "data-bucket"
            }
        },
        "interface": ":4984",
        "log": [
            "CRUD+",
            "Cache+",
            "HTTP+",
            "Changes+",
            "Import+"
        ],
        "maxCouchbaseConnections": 16,
        "maxFileDescriptors": 90000,
        "maxIncomingConnections": 0,
        "slowServerCallWarningThreshold": 500
    }
```

### Log output
https://gist.github.com/sethrosetter/a34bcee11e189c914147222839c0e77d

### Expected behavior
Backfill should send all changes

### Actual behavior
Backfill only sending a subset

Failing test - http://uberjenkins.sc.couchbase.com/job/cen7-sync-gateway-functional-tests-base-cc/873/testReport/junit/testsuites.syncgateway.functional.tests/test_views/test_view_backfill_for_deletes_sync_gateway_default_functional_tests_False_/

**Comment by sethrosetter** [comment]
docmeta successful run: https://gist.github.com/sethrosetter/c4d8a9798746925a640a3aec521b28c6

**Comment by sethrosetter** [comment]
Run with xattrs and 1.5.0-475 succeeds, https://gist.github.com/sethrosetter/4adb1b7d47b33faba8baa0853cd4f84b

**Comment by adamcfraser** [comment]
This appears to be a view engine issue that's been exposed by Sync Gateway's recent uptake of server support for a single-op tombstone (delete body and update xattr in a single operation).

Filed https://issues.couchbase.com/browse/MB-25422

**Comment by adamcfraser** [comment]
Addressed for 1.5 beta refresh by https://github.com/couchbase/sync_gateway/pull/2744.

Leaving this ticket open for re-enabling single-op delete after 1.5 beta is cut - can re-enable now that https://issues.couchbase.com/browse/MB-25422 is fixed for the latest Spock builds.

**Comment by djpongh** [comment]
Should address #2627
